core/domain/user_jobs_one_off_test.py

```diff
@@ -1138,3 +1139,70 @@ def test_that_last_edited_and_created_time_are_not_updated(self):
         owner_settings = user_services.get_user_settings(self.owner_id)
         self.assertIsNone(owner_settings.last_created_an_exploration)
         self.assertIsNone(owner_settings.last_edited_an_exploration)
+
+
+class UserRolesMigrationOneOffJobTests(test_utils.GenericTestBase):
+
+    def test_create_user_with_different_roles_and_run_migration_job(self):
```

Comment:
It's good practice to add a docstring here describing what the test does, since the test is quite long.